src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -3622,7 +3622,7 @@ bool IdealLoopTree::do_remove_empty_loop(PhaseIdealLoop *phase) {
   Node* iv = NULL;
   for (DUIterator_Fast imax, i = cl->fast_outs(imax); i < imax; i++) {
     Node* n = cl->fast_out(i);
-    if (n->Opcode() == Op_Phi) {
+    if ((n->Opcode() == Op_Phi) && (n->outcnt() > 0)) {
```

Comment:
May be add comment to explain additional check.

Reply:
@vnkozlov thanks for the suggestion: I've updated the existing comment instead (I hope it is clear enough).